packages/@react-aria/collections/src/Document.ts

```diff
@@ -290,47 +293,59 @@ export class ElementNode<T> extends BaseNode<T> {
    */
   private getMutableNode(): Mutable<CollectionNode<T>> {
     if (!this.isMutated) {
-      this.node = this.node.clone();
+      this.node = this.node!.clone();
```

Comment:
this isn't actually true, and if the assumption ever doesn't hold, then a guard would be useful with console warning as to what likely went wrong